gtsfm/frontend/verifier/gric_verifier.py

```diff
@@ -0,0 +1,191 @@
+"""
+Wrapper about COLMAP's GRIC estimators, using pycolmap's pybind API.
+Uses LORANSAC for Essential, Fundamental, and Homography matrix estimation.
+
+LORANSAC paper:
+ftp://cmp.felk.cvut.cz/pub/cmp/articles/matas/chum-dagm03.pdf
+
+On Linux and Mac, a python wheel is available:
+https://pypi.org/project/pycolmap/#files
+
+Authors: John Lambert
+"""
+
+from enum import Enum
+from typing import Any, Dict, Optional, Tuple
+
+import numpy as np
+import pycolmap
+from gtsam import Cal3Bundler, Rot3, Unit3
+
+import gtsfm.frontend.verifier.verifier_base as verifier_base
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.verifier.verifier_base import VerifierBase
+
+
+logger = logger_utils.get_logger()
+
+
+MIN_INLIER_RATIO = 0.01
+MIN_NUM_TRIALS = 100000
+MAX_NUM_TRIALS = 1000000
+CONFIDENCE = 0.999999
+
+
+class ConfigurationType(Enum):
+    UNDEFINED = 0
+    # Degenerate configuration (e.g., no overlap or not enough inliers).
+    DEGENERATE = 1
+    # Essential matrix.
+    CALIBRATED = 2
+    # Fundamental matrix.
+    UNCALIBRATED = 3
+    # Homography, planar scene with baseline.
+    PLANAR = 4
+    # Homography, pure rotation without baseline.
+    PANORAMIC = 5
+    # Homography, planar or panoramic.
+    PLANAR_OR_PANORAMIC = 6
+    # Watermark, pure 2D translation in image borders.
+    WATERMARK = 7
+    # Multi-model configuration, i.e. the inlier matches result from multiple
+    # individual, non-degenerate configurations.
+    MULTIPLE = 8
+
+
+class GricVerifier(VerifierBase):
+    def __init__(
+        self,
+        use_intrinsics_in_verification: bool,
+        estimation_threshold_px: float,
+    ) -> None:
+        """Initializes the verifier.
+
+        (See https://github.com/mihaidusmanu/pycolmap/blob/master/essential_matrix.cc#L98)
+
+        Args:
+            use_intrinsics_in_verification: Flag to perform keypoint normalization and compute the essential matrix
+                instead of fundamental matrix. This should be preferred when the exact intrinsics are known.
+            estimation_threshold_px: maximum distance (in pixels) to consider a match an inlier, under squared
+                Sampson distance.
+        """
+        self._use_intrinsics_in_verification = use_intrinsics_in_verification
```

Comment:
Are we switching between E/F estimation?

Reply:
GRIC automatically checks E. vs. F vs. H

Reply:
(I just added a comment to explain that better)